src/main/kotlin/com/curiouscreature/kotlin/math/Matrix.kt

```diff
@@ -386,7 +386,78 @@ fun transpose(m: Mat4) = Mat4(
+    pairs[7]  = m.w.z * m.x.w
+    pairs[8]  = m.x.z * m.z.w
+    pairs[9]  = m.z.z * m.x.w
+    pairs[10] = m.x.z * m.y.w
+    pairs[11] = m.y.z * m.x.w
+
+    result.x.x = pairs[0] * m.y.y + pairs[3] * m.z.y + pairs[4] * m.w.y
+    result.x.x -= pairs[1] * m.y.y + pairs[2] * m.z.y + pairs[5] * m.w.y
+    result.x.y = pairs[1] * m.x.y + pairs[6] * m.z.y + pairs[9] * m.w.y
+    result.x.y -= pairs[0] * m.x.y + pairs[7] * m.z.y + pairs[8] * m.w.y
+    result.x.z = pairs[2] * m.x.y + pairs[7] * m.y.y + pairs[10] * m.w.y
+    result.x.z -= pairs[3] * m.x.y + pairs[6] * m.y.y + pairs[11] * m.w.y
+    result.x.w = pairs[5] * m.x.y + pairs[8] * m.y.y + pairs[11] * m.z.y
+    result.x.w -= pairs[4] * m.x.y + pairs[9] * m.y.y + pairs[10] * m.z.y
+    result.y.x = pairs[1] * m.y.x + pairs[2] * m.z.x + pairs[5] * m.w.x
+    result.y.x -= pairs[0] * m.y.x + pairs[3] * m.z.x + pairs[4] * m.w.x
+    result.y.y = pairs[0] * m.x.x + pairs[7] * m.z.x + pairs[8] * m.w.x
+    result.y.y -= pairs[1] * m.x.x + pairs[6] * m.z.x + pairs[9] * m.w.x
+    result.y.z = pairs[3] * m.x.x + pairs[6] * m.y.x + pairs[11] * m.w.x
+    result.y.z -= pairs[2] * m.x.x + pairs[7] * m.y.x + pairs[10] * m.w.x
+    result.y.w = pairs[4] * m.x.x + pairs[9] * m.y.x + pairs[10] * m.z.x
+    result.y.w -= pairs[5] * m.x.x + pairs[8] * m.y.x + pairs[11] * m.z.x
+
+    pairs[0] = m.z.x * m.w.y
+    pairs[1] = m.w.x * m.z.y
+    pairs[2] = m.y.x * m.w.y
+    pairs[3] = m.w.x * m.y.y
+    pairs[4] = m.y.x * m.z.y
+    pairs[5] = m.z.x * m.y.y
+    pairs[6] = m.x.x * m.w.y
+    pairs[7] = m.w.x * m.x.y
+    pairs[8] = m.x.x * m.z.y
+    pairs[9] = m.z.x * m.x.y
+    pairs[10] = m.x.x * m.y.y
+    pairs[11] = m.y.x * m.x.y
+
+    result.z.x = pairs[0] * m.y.w + pairs[3] * m.z.w + pairs[4] * m.w.w
+    result.z.x -= pairs[1] * m.y.w + pairs[2] * m.z.w + pairs[5] * m.w.w
+    result.z.y = pairs[1] * m.x.w + pairs[6] * m.z.w + pairs[9] * m.w.w
+    result.z.y -= pairs[0] * m.x.w + pairs[7] * m.z.w + pairs[8] * m.w.w
+    result.z.z = pairs[2] * m.x.w + pairs[7] * m.y.w + pairs[10] * m.w.w
+    result.z.z -= pairs[3] * m.x.w + pairs[6] * m.y.w + pairs[11] * m.w.w
+    result.z.w = pairs[5] * m.x.w + pairs[8] * m.y.w + pairs[11] * m.z.w
+    result.z.w -= pairs[4] * m.x.w + pairs[9] * m.y.w + pairs[10] * m.z.w
+    result.w.x = pairs[2] * m.z.z + pairs[5] * m.w.z + pairs[1] * m.y.z
+    result.w.x -= pairs[4] * m.w.z + pairs[0] * m.y.z + pairs[3] * m.z.z
+    result.w.y = pairs[8] * m.w.z + pairs[0] * m.x.z + pairs[7] * m.z.z
+    result.w.y -= pairs[6] * m.z.z + pairs[9] * m.w.z + pairs[1] * m.x.z
+    result.w.z = pairs[6] * m.y.z + pairs[11] * m.w.z + pairs[3] * m.x.z
+    result.w.z -= pairs[10] * m.w.z + pairs[2] * m.x.z + pairs[7] * m.y.z
+    result.w.w = pairs[10] * m.z.z + pairs[4] * m.x.z + pairs[9] * m.y.z
+    result.w.w -= pairs[8] * m.y.z + pairs[11] * m.z.z + pairs[5] * m.x.z
+
+    val det = 1 / (m.x.x * result.x.x + m.y.x * result.x.y + m.z.x * result.x.z + m.w.x * result.x.w)
```

Comment:
Call it determinant. Also use 1.0f / instead of 1 /.  prefer to avoid mixing integers and floats for clarity.